src/webui/api/torrentcreatorcontroller.cpp

```diff
@@ -0,0 +1,247 @@
+    }
+
+    QString torrentFormatToString(const BitTorrent::TorrentFormat torrentFormat)
+    {
+        switch (torrentFormat)
+        {
+        case BitTorrent::TorrentFormat::V1:
+            return u"v1"_s;
+        case BitTorrent::TorrentFormat::V2:
+            return u"v2"_s;
+        default:
+            return u"hybrid"_s;
+        }
+    }
+#endif
+
+    QString taskStatusString(const std::shared_ptr<BitTorrent::TorrentCreationTask> task)
+    {
+        if (task->isFailed())
+            return u"Failed"_s;
+
+        switch (task->state())
+        {
+        case BitTorrent::TorrentCreationTask::Queued:
+        default:
+            return u"Queued"_s;
+        case BitTorrent::TorrentCreationTask::Running:
+            return u"Running"_s;
+        case BitTorrent::TorrentCreationTask::Finished:
+            return u"Finished"_s;
+        }
+    }
+}
+
+TorrentCreatorController::TorrentCreatorController(BitTorrent::TorrentCreationManager *torrentCreationManager, IApplication *app, QObject *parent)
+    : APIController(app, parent)
+    , m_torrentCreationManager {torrentCreationManager}
+{
+}
+
+void TorrentCreatorController::addTaskAction()
+{
+    requireParams({KEY_SOURCE_PATH});
+
+    const BitTorrent::TorrentCreatorParams createTorrentParams
+    {
+        .isPrivate = parseBool(params()[KEY_PRIVATE]).value_or(false),
+#ifdef QBT_USES_LIBTORRENT2
+        .torrentFormat = parseTorrentFormat(params()[KEY_FORMAT].toLower()),
+#else
+        .isAlignmentOptimized = parseBool(params()[KEY_OPTIMIZE_ALIGNMENT]).value_or(true),
+        .paddedFileSizeLimit = parseInt(params()[KEY_PADDED_FILE_SIZE_LIMIT]).value_or(-1),
+#endif
+        .pieceSize = parseInt(params()[KEY_PIECE_SIZE]).value_or(0),
+        .sourcePath = Path(params()[KEY_SOURCE_PATH]),
+        .torrentFilePath = Path(params()[KEY_TORRENT_FILE_PATH]),
+        .comment = params()[KEY_COMMENT],
+        .source = params()[KEY_COMMENT],
+        .trackers = params()[KEY_TRACKERS].split(u'|'),
+        .urlSeeds = params()[KEY_URL_SEEDS].split(u'|')
+    };
+
+    bool const startSeeding = parseBool(params()[u"startSeeding"_s]).value_or(createTorrentParams.torrentFilePath.isEmpty());
+
+    auto task = m_torrentCreationManager->createTask(createTorrentParams, startSeeding);
+    if (!task)
+        throw APIError(APIErrorType::Conflict, tr("Too many active tasks"));
+
+    setResult(QJsonObject {{KEY_TASK_ID, task->id()}});
+}
+
+void TorrentCreatorController::statusAction()
+{
+    const QString id = params()[KEY_TASK_ID];
+
+    const auto singleTask = m_torrentCreationManager->getTask(id);
+    if (!id.isEmpty() && !singleTask)
+        throw APIError(APIErrorType::NotFound);
+
+    const QList<std::shared_ptr<BitTorrent::TorrentCreationTask>> tasks = id.isEmpty()
+            ? m_torrentCreationManager->tasks()
+            : QList<std::shared_ptr<BitTorrent::TorrentCreationTask>> {singleTask};
+    QJsonArray statusArray;
```

Comment:
Wouldn't it be better to return them as a map (JSON object) with task IDs as keys?

Reply:
> Wouldn't it be better to return them as a map (JSON object) with task IDs as keys?

@rcarpa 
What do you think about this?